backend/models/user.js

```diff
@@ -1068,155 +928,98 @@ function _createAccounts(roles, userName) {
-
-	let foundProjects = [];
-	let foundModels = [];
-
-	if (this.user === username) {
+User.removeTeamMember = async function (teamspace, userToRemove, cascadeRemove) {
+	if (teamspace.user === userToRemove) {
 		// The user should not be able to remove itself from the teamspace
 		return Promise.reject(responseCodes.SUBSCRIPTION_CANNOT_REMOVE_SELF);
 	}
 
-	const teamspacePerm = this.customData.permissions.findByUser(username);
-	// check if they have any permissions assigned
-	return Project.find({ account: this.user }, { "permissions.user": username }).then(projects => {
-
-		foundProjects = projects;
-		return ModelSetting.find({ account: this.user }, { "permissions.user": username });
+	const teamspacePerm = AccountPermissions.findByUser(teamspace, userToRemove);
 
-	}).then(models => {
+	// check if they have any permissions assigned
+	const [projects, models] = await Promise.all([
+		Project.find({ account: teamspace.user }, { "permissions.user": userToRemove }),
+		ModelSetting.find({ account: teamspace.user }, { "permissions.user": userToRemove })
+	]);
 
-		foundModels = models;
+	if (!cascadeRemove && (models.length || projects.length || teamspacePerm)) {
+		throw({
+			resCode: responseCodes.USER_IN_COLLABORATOR_LIST,
+			info: {
+				models: models.map(m => {
+					return { model: m.name };
+				}),
+				projects: projects.map(p => p.name),
+				teamspace: teamspacePerm
+			}
+		});
+	} else {
 
-		if (!cascadeRemove && (foundModels.length || foundProjects.length || teamspacePerm)) {
+		const promises = [];
 
-			return Promise.reject({
-				resCode: responseCodes.USER_IN_COLLABORATOR_LIST,
-				info: {
-					models: foundModels.map(m => {
-						return { model: m.name };
-					}),
-					projects: foundProjects.map(p => p.name),
-					teamspace: teamspacePerm
-				}
-			});
+		if (teamspacePerm) {
+			promises.push(AccountPermissions.remove(teamspace, userToRemove));
+		}
 
-		} else {
+		promises.push(models.map(model =>
+			model.changePermissions(model.permissions.filter(p => p.user !== userToRemove))));
 
-			const promises = [];
+		promises.push(projects.map(project =>
+			project.updateAttrs({ permissions: project.permissions.filter(p => p.user !== userToRemove) })));
 
-			if (teamspacePerm) {
-				promises.push(this.customData.permissions.remove(username));
-			}
+		promises.push(removeUserFromAnyJob(teamspace.user, userToRemove));
 
-			promises.push(foundModels.map(model =>
-				model.changePermissions(model.permissions.filter(p => p.user !== username))));
+		await Promise.all(promises);
+	}
 
-			promises.push(foundProjects.map(project =>
-				project.updateAttrs({ permissions: project.permissions.filter(p => p.user !== username) })));
+	return await Role.revokeTeamSpaceRoleFromUser(userToRemove, teamspace.user);
+};
 
-			promises.push(removeUserFromAnyJob(this.user, username));
-			return Promise.all(promises);
-		}
+User.addTeamMember = async function(teamspace, userToAdd, job, permissions) {
```

Comment:
Can we also make this pass in a teamspace name instead of object?

This is used in invitations.js and routes/, both of which are currently fetching the user object just to call this function